src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -234,8 +234,8 @@ const std::shared_ptr<SQLQueryPlan>& SQLPipelineStatement::get_query_plan() {
     _prepared_statements->set(prepared_statement->name, *_query_plan);
   }
 
-  // Cache newly created plan for the according sql statement (only if not already cached)
-  if (!_query_plan_cache_hit) {
+  // Cache newly created plan for the according sql statement (only if not already cached or is not data definition)
+  if (!_query_plan_cache_hit && !statement->isDataDefinitionStatement()) {
```

Comment:
After giving this some more thought (see my comment), I feel like this should check if `statement` is a `SELECT` statement. While this means that more complicated inserts like `INSERT INTO ... SELECT (some long statement)` do not get cached, this is probably the easier way.